protostar/profiler/transaction_profiler_test.py

```diff
@@ -0,0 +1,223 @@
+    build_global_instructions,
+    get_instruction_id_offsets,
+    translate_callstack,
+)
+
+from protostar.starknet.cheatable_execute_entry_point import ContractProfile
+
+profile = RuntimeProfile(
+    [Function("f1", "d", 1), Function("f2", "d", 2)], [], [], [], []
+)
+
+
+@pytest.mark.parametrize(
+    "samples,result",
+    [
+        (
+            [
+                ContractProfile(["a", "b"], ContractEntryPoint(0, 0), replace(profile)),
+                ContractProfile(["c", "b"], ContractEntryPoint(0, 0), replace(profile)),
+            ],
+            {
+                "b.f1": replace(profile.functions[0], id="b.f1"),
+                "b.f2": replace(profile.functions[1], id="b.f2"),
+            },
+        ),
+        ([], {}),
+        (
+            [
+                ContractProfile(["a", "b"], ContractEntryPoint(0, 0), replace(profile)),
+                ContractProfile(["a", "c"], ContractEntryPoint(0, 0), replace(profile)),
+            ],
+            {
+                "b.f1": replace(profile.functions[0], id="b.f1"),
+                "b.f2": replace(profile.functions[1], id="b.f2"),
+                "c.f1": replace(profile.functions[0], id="c.f1"),
+                "c.f2": replace(profile.functions[1], id="c.f2"),
+            },
+        ),
+    ],
+)
+def test_build_global_functions(samples, result):
+    assert build_global_functions(samples) == result
+
+
+funct = Function("z", "z", 0)
+ep = ContractEntryPoint(0, 0)
+
+
+@pytest.mark.parametrize(
+    "samples,result",
+    [
+        (
+            [
+                ContractProfile(
+                    ["a"],
+                    ep,
+                    RuntimeProfile(
+                        [],
+                        [
+                            Instruction(1000, 0, replace(funct), 0),
+                            Instruction(100, 0, replace(funct), 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
+                ContractProfile(
+                    ["b"],
+                    ep,
+                    RuntimeProfile(
+                        [],
+                        [
+                            Instruction(150, 0, replace(funct), 0),
+                            Instruction(100, 0, replace(funct), 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
+                ContractProfile(
+                    ["c"],
+                    ep,
+                    RuntimeProfile(
+                        [],
+                        [
+                            Instruction(10, 0, replace(funct), 0),
+                            Instruction(100, 0, replace(funct), 0),
+                        ],
+                        [],
+                        [],
+                        [],
+                    ),
+                ),
```

Comment:
This going to be hard to maintain. Can you hide this somehow behind a readible testing API?

Reply:
I will think about it, but it is gonna be hard. It looked much better before linting 😓

Reply:
Removing the trailing comma might help.